### PR TITLE
docs:add two usercase blogs

### PR DESCRIPTION
**Pull request by KID-G**
Changes:

add two usercase and one Tongsuo blogs.

### CODE REVIEW

File: blog/zh/blog/2022/12/08/apisix-support-tongsuo.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+因此，当我们谈论国密支持时，并不仅仅单独指可以用某一种国密算法进行加解密，而是指嵌合入国密的生态，支持某种国密的应用场景。
+
+## 国密的应用场景
+
+作为国家密码管理局制定的密码算法，国密广泛应用于电子政务（包括国家政务通、警务通等重要领域）、信创及金融业的各个应用领域。
+
+* **政府和金融的身份认证终端**。依照现行有关规定，许多涉及政府和金融的身份认证终端（诸如 USBKey、智能 IC 卡、银行卡终端等）都需要提供对国密的支持。
+* **国产开源操作系统**。许多主打国产替代的开源操作系统，会提供基于国密的安全加固功能。比如龙蜥操作系统 (Anolis OS) 提到自己实现了全栈国密能力；OpenEuler 也在做国密相关的一些功能，比如基于国密数字证书扩展了 EFI 的数字签名。
+* **信创产品**。还有许多做信创生意的厂商，围绕国密推出符合相关标准的产品。例如支持使用国密算法做数字签名的 PDF 工具、支持国密接入标准的音视频软件等等。
+* **基于国密 TLS 协议的生态**。也是在日常开发中接触得最多的。譬如各种国产 CA 厂商、支持了国密TLS的许许多多密码库和浏览器，以及国密接入的VPN和网关等等。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/08/apisix-support-tongsuo.md]:
```suggestion
* **基于国密 TLS 协议的生态**。也是在日常开发中接触得最多的。譬如各种国产 CA 厂商、支持了国密 TLS 的许许多多密码库和浏览器，以及国密接入的 VPN 和网关等等。
```

File: blog/zh/blog/2022/12/08/apisix-support-tongsuo.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+Apache APISIX 是一个动态、实时、高性能的 API 网关，提供负载均衡、动态上游、灰度发布、精细化路由、限流限速、服务降级、服务熔断、身份认证、可观测性等数百项功能。作为一个发迹于国内环境的 API 网关，APISIX 自然需要考虑下如何接入国密的生态圈。
+
+由于国密更多地用在国内环境中，尚未被 OpenSSL 等国际主流项目所完全接纳。如果要使用国密的功能，就涉及到更换 APISIX 默认的 OpenSSL 库为其他 SSL 库。
+
+为此，我们考察了以下的项目：
+
+* GMSSL：北京大学开源的项目。原版基于 OpenSSL 1.0.2 修改而来。新的 GMSSL 3.0 基本上是重新开发，跟 OpenSSL 的目录结构差别很大。
+* gm-BoringSSL：个人开源项目，在 BoringSSL 上增加国密支持。已有两年未改动。
+* TaSSL：北京江南天安科技有限公司开源的项目。基于 OpenSSL 1.1.1 修改而来。
+* Tongsuo：蚂蚁集团开源的项目。基于 OpenSSL 3.0 修改而来，项目前身是 BabaSSL，现已 **改名为铜锁/Tongsuo**。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/08/apisix-support-tongsuo.md]:
```suggestion
* Tongsuo：蚂蚁集团开源的项目。基于 OpenSSL 3.0 修改而来，项目前身是 BabaSSL，现已**改名为铜锁/Tongsuo**。
```

File: blog/zh/blog/2022/12/08/apisix-support-tongsuo.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+由于 GMSSL 3.0 并不基于 OpenSSL，即使能保证 API 兼容，也没办法确保能 100% 替换现有 OpenSSL 的行为，所以被首先排除。其次 gm-BoringSSL 疏于维护，也被排除。
+
+在 TaSSL 和 Tongsuo 之中，我倾向于选择 Tongsuo[1]。因为 Tongsuo 在标准上拥有更强的话语权，比如 RFC 8998（TLS 1.3 中支持 SM 套件）就是由 Tongsuo 的开发者制定的。TaSSL 则是每出一个版本，就公布一个新的仓库。比如前一个 版本[2]，感觉不太靠谱。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/08/apisix-support-tongsuo.md]:
```suggestion
在 TaSSL 和 Tongsuo 之中，我倾向于选择 [Tongsuo](https://github.com/Tongsuo-Project/Tongsuo)。因为 Tongsuo 在标准上拥有更强的话语权，比如 RFC 8998（TLS 1.3 中支持 SM 套件）就是由 Tongsuo 的开发者制定的。TaSSL 则是每出一个版本，就公布一个新的仓库。比如[前一个 版本](https://github.com/jntass/TASSL-1.1.1k)，感觉不太靠谱。
```

File: blog/zh/blog/2022/12/08/apisix-support-tongsuo.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+对于选择 Tongsuo，我个人存在一个顾虑，就是他目前基于 OpenSSL 3.0 的版本还没有发布正式的 Release 版本（第一个版本预计在 2023 年 2 月发布）。由于 Tongsuo 当前还没有一个固定的版本，因此社区决定先把国密相关的功能独立出来，以插件形式存在，有相关需求时可单独启用。
+
+目前已在插件层面实现了服务端一侧国密双证书的支持，感兴趣的读者可以在官网查看  `gm` 插件介绍文档 插件介绍文档[3]，自行完成 APISIX 的编译和对应插件的安装配置工作。当然，如果想即刻预览该插件的使用过程，也可以直接参考下文内容。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/08/apisix-support-tongsuo.md]:
```suggestion
目前已在插件层面实现了服务端一侧国密双证书的支持，感兴趣的读者可以在官网查看  `gm` [插件介绍文档](https://apisix.apache.org/zh/docs/apisix/next/plugins/gm/)，自行完成 APISIX 的编译和对应插件的安装配置工作。当然，如果想即刻预览该插件的使用过程，也可以直接参考下文内容。
```

File: blog/zh/blog/2022/12/08/apisix-support-tongsuo.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    ...
+    # 可按实际情况调整。错误的 cipher 会导致 “no shared cipher” 或 “no ciphers available” 报错。
+    ssl_ciphers: HIGH:!aNULL:!MD5
+```
+
+配置完成后，重新加载 APISIX，此时 APISIX 将会启用国密相关的逻辑。
+
+## 测试插件
+
+在测试插件之前，需要准备好国密双证书。Tongsuo 提供了生成 【SM2 双证书】 的 教程[4]。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/08/apisix-support-tongsuo.md]:
```suggestion
在测试插件之前，需要准备好国密双证书。Tongsuo 提供了[生成 SM2 双证书](https://www.yuque.com/tsdoc/ts/sulazb)的教程。
```

File: blog/zh/blog/2022/12/08/apisix-support-tongsuo.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+如果你对该功能或者插件感兴趣，欢迎在随时在社区进行交流。
+
+[1] [https://github.com/Tongsuo-Project/Tongsuo](https://github.com/Tongsuo-Project/Tongsuo)
+
+[2] [https://github.com/jntass/TASSL-1.1.1k](https://github.com/jntass/TASSL-1.1.1k)
+
+[3] [https://apisix.apache.org/zh/docs/apisix/next/plugins/gm/](https://apisix.apache.org/zh/docs/apisix/next/plugins/gm/)
+
+[4] [https://www.yuque.com/tsdoc/ts/sulazb](https://www.yuque.com/tsdoc/ts/sulazb)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/12/08/apisix-support-tongsuo.md]: delete this part

Comment (KID-G) [reply blog/zh/blog/2022/12/08/apisix-support-tongsuo.md]: LGTM about all

**Review by SylviaBABY — DISMISSED** [review]

File: blog/zh/blog/2022/12/07/junrunrenli-with-apisix.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+![君润人力业务](https://static.apiseven.com/2022/12/09/639291b9ce2a8.png)
+
+## 君润人力自研系统架构概述
+
+君润人力在搭建自研服务平台架构时，首要原则就是“开放、开源、拥抱云原生”。平台基于 K8s 微服务架构构建云端系统，整体架构参考下图。
+
+![自研服务平台](https://static.apiseven.com/2022/12/09/639291ba824c6.png)
+
+可以看到，君润自研服务平台 DevOps 流程（右侧）依托 Git webhook、coding、K8s 集群实现全自动化与无感发布，业务系统基于 PaaS 平台进行迭代开发，确保研发规范，并达成技术栈的统一。（上侧）在监控手段上，系统集成了 sky-agent 和 arthas，满足了线上服务观测多维度的需求。与此同时，采用腾讯日志云与 K8s 结合的方式将服务日志存储在云上，实现 K8s 集群内无文件化服务，避免磁盘容量堆积从而影响到 K8s 集群单个节点的健康。
```
Comment (SkyeYoung) [inline blog/zh/blog/2022/12/07/junrunrenli-with-apisix.md]:
```suggestion
可以看到，君润自研服务平台 DevOps 流程（右侧）依托 Git webhook、CODING、K8s 集群实现全自动化与无感发布，业务系统基于 PaaS 平台进行迭代开发，确保研发规范，并达成技术栈的统一。（上侧）在监控手段上，系统集成了 sky-agent 和 arthas，满足了线上服务观测多维度的需求。与此同时，采用腾讯日志云与 K8s 结合的方式将服务日志存储在云上，实现 K8s 集群内无文件化服务，避免磁盘容量堆积从而影响到 K8s 集群单个节点的健康。
```

Does coding refer to Tencent Cloud's CODING? If so, you can consider committing the suggestion

File: blog/zh/blog/2022/12/07/junrunrenli-with-apisix.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+架构图的左侧，是业务系统中最重要的环节 -- 流量管理。所有流量会通过 WAF（WAF，Web应用防火墙，负责网络安全）进入三层网络里面的第一层 CLB，它主要负责流量的转发；第二层就是云原生网关 APISIX，它承担了业务系统的内部服务治理；第三层则是业务系统的 Gateway 应用内部的网关，负责单系统鉴权和路由。
```
Comment (SkyeYoung) [inline blog/zh/blog/2022/12/07/junrunrenli-with-apisix.md]:
```suggestion
架构图的左侧，是业务系统中最重要的环节 -- 流量管理。所有流量会通过 WAF（Web Application Firewall，Web应用防火墙，负责网络安全）进入三层网络里面的第一层 CLB，它主要负责流量的转发；第二层就是云原生网关 APISIX，它承担了业务系统的内部服务治理；第三层则是业务系统的 Gateway 应用内部的网关，负责单系统鉴权和路由。
```

File: blog/zh/blog/2022/12/07/junrunrenli-with-apisix.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+通过该架构图不难发现，第二层是所有流量的入口，选择一个满足业务扩张需求的云原生网关，对我们来说至关重要，下面谈谈在网关技术选型时的主要思考。
+
+## 技术网关选型痛点
+
+- 数量庞大的业务系统。我们在人力资源这个领域有 15+ 服务平台，生态业务多样化，面临的问题也比较多，服务请求需要频繁变更，导致需要操作和配置的路由也非常多。原来我们是基于 CLB 进行流量转发，久而久之，运维人员需要配置和操作的地方非常多，耗费了大量的人力与时间。
+
+- 频繁的高并发大流量。举个例子，客户集中在同一天发薪或提现大额资金。在用户数量达到大几十万时，集中进行的某些行为，如打卡、签约、领取任务或工资等，此时系统并发流量非常大，短期翻倍的情况比比皆是。
+
+- 个性化需求的扩张压力。APISIX 是基于 OpenResty、Nginx 和 Lua 开发的，但如果用 Lua 来开发插件，会有一定的研发投入和维护成本。
```
Comment (SkyeYoung) [inline blog/zh/blog/2022/12/07/junrunrenli-with-apisix.md]:
```suggestion
- 个性化需求的扩张压力。APISIX 是基于 OpenResty、NGINX 和 Lua 开发的，但如果用 Lua 来开发插件，会有一定的研发投入和维护成本。
```

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Review by SkyeYoung — APPROVED** [review]

**Pull request merged**